### PR TITLE
Adds support for clients and client credentials.  Closes #611.

### PR DESCRIPTION
**Pull request by 345andrew**

### CODE REVIEW

**Comment by yaythomas** [comment]
You didn't mean master, presumably.

**Comment by 345andrew** [comment]
Reopening - master was correct branch

File: cloco_cli/cli.py [hunk 1]
```diff
@@ -168,6 +168,111 @@ def delete_subscription_permission(sub, username):
...
+@click.option('--sub', help='The subscription identifier, will use the subscription stored in the preferences if not supplied', default='')
+@click.option('--name', help='The username of the client')
+@click.option('--key', help='The client key of the credentials to be deleted')
+def delete_client_credentials(sub, name, key):
+    """Deletes client credentials."""
+    config = load_config()
+    authenticate(config)
+    if not sub:
+        sub = config['preferences']['subscription']
+    u = '{0}/{1}/clients/{2}/credentials/{3}'.format(
```
Comment (yaythomas) [inline cloco_cli/cli.py:269]: not important, but fyi, you could use f-strings for something like `f'{url}/blah/clients/credentials/{1}'` depending on where you think readability is better

File: cloco_cli/cli.py [hunk 2]
```diff
@@ -168,6 +168,111 @@ def delete_subscription_permission(sub, username):
...
+@click.option('--sub', help='The subscription identifier, will use the subscription stored in the preferences if not supplied', default='')
+def list_subscription_client(sub):
+    """Returns a list of the subscription clients"""
+    config = load_config()
+    authenticate(config)
+    if not sub:
+        sub = config['preferences']['subscription']
+    u = '{0}/{1}/clients'.format(get_url(config), sub)
+    r = requests.get(u, headers=get_headers(config))
+    print_json_response(r)
```
Comment (yaythomas) [inline cloco_cli/cli.py:187]: more of a general question, response contain sensitive things like api key or somesuch when it dumps to stdout?

**Review by yaythomas — APPROVED** [review]
:shipit:

File: cloco_cli/cli.py (hunk 1 above)
Comment (yaythomas) [reply cloco_cli/cli.py:269]: Well, and how backwards compatible you want it. I think f strings is only 3.6

**Pull request merged**